pkg/plugins/helper/node_info.go

```diff
@@ -0,0 +1,104 @@
+/*
+Copyright 2024 The Godel Scheduler Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package helper
+
+import (
+	"fmt"
+
+	"github.com/kubewharf/godel-scheduler/pkg/binder/framework/handle"
+	framework "github.com/kubewharf/godel-scheduler/pkg/framework/api"
+	podutil "github.com/kubewharf/godel-scheduler/pkg/util/pod"
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+const (
+	AllNodeInfoKey = "AllNodeInfo"
+)
+
+type AllNodeInfos struct {
+	NodeInfos []framework.NodeInfo
+}
+
+func (f AllNodeInfos) Clone() framework.StateData {
+	clonedNodeInfos := make([]framework.NodeInfo, len(f.NodeInfos))
+
+	copy(clonedNodeInfos, f.NodeInfos)
+
+	return AllNodeInfos{
+		NodeInfos: clonedNodeInfos,
+	}
+}
+
+func GetAllNodeInfos(frameworkHandle handle.BinderFrameworkHandle) (*AllNodeInfos, error) {
+	nodeInfoMap := map[string]framework.NodeInfo{}
+	for _, podLauncher := range podutil.PodLanucherTypes {
+		if podLauncher == podutil.Kubelet && frameworkHandle.SharedInformerFactory() != nil {
+			allV1Nodes, err := frameworkHandle.SharedInformerFactory().Core().V1().Nodes().Lister().List(labels.Everything())
+			if err != nil {
+				return nil, err
+			}
+
+			for _, node := range allV1Nodes {
+				nodeInfo := frameworkHandle.GetNodeInfo(node.Name)
```

Comment:
NodeInfo may be nil

Reply:
This function is no longer used. I have deleted it.